tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -504,27 +579,386 @@ impl<
+							},
+							_ => {
+								log::debug!("Binance order status: {:?}", trade_order.status);
+							},
+						}
+						//todo: how long we wait ?
+						sleep(Duration::from_millis(500)).await;
+					}
+					if !trade_success {
+						log::error!("Binance order failed");
+						let data = CrossOrderFailData {
+							request_id: intent_id,
+							// TODO: is this a user facing error? what should we return?
+							fail_reason: "Binance order failed".to_string(),
+						};
+						self.pumpx_api.cross_fail(&access_token, data).await.map_err(|_| {
+							log::error!("Failed to notify pumpx-signer");
+						})?;
+						// TODO: Figure out how to transfer back the asset to the omni account
+						// check https://developers.binance.com/docs/wallet/capital/withdraw
+
+						return Err(());
+					}
+
+					debug!("Total traded on binance: {:?}", to_amount);
+
+					debug!("Getting {:?} nonce for payout request", payout_address);
+					// 4. Call accounting contract on BSC
+					let user_nonce =
+						self.accounting_contract_client.get_nonce(payout_address).await.map_err(
+							|_| {
+								log::error!("Failed to get nonce");
+							},
+						)?;
+
+					debug!("Received {:?} nonce", user_nonce);
+
+					debug!("Calling accounting contract payout with address {:?}, nonce {:?} and amount {:?}", payout_address, user_nonce, to_amount);
+					//todo : make sure we payout correct amount
+					self.accounting_contract_client
+						.execute_pay_out_request(
+							payout_address,
+							user_nonce,
+							U256::from_str(&to_amount.to_string()).unwrap(),
+						)
+						.await
+						.map_err(|_| {
+							log::error!("Failed to execute pay out request");
+						})?;
+
+					let Some(chain_type) = ChainType::from_pumpx_chain_id(pumpx_config.to_chain_id)
+					else {
+						log::error!("Unsupported to_chain_id: {}", pumpx_config.to_chain_id);
+						return Err(());
+					};
+
+					let wallet_address: String = match chain_type {
+						ChainType::Evm => match swap_order.to_address.unwrap() {
+							HeimaMultiAddress::Address20(address) => {
+								hex_encode_evm_address_bytes(address.as_ref())
+							},
+							_ => {
+								log::error!(
+									"Wrong address: {:?}, expected Address20",
+									swap_order.to_address
+								);
+								return Err(());
+							},
+						},
+						_ => {
+							log::error!("Chain type not supported: {:?}", chain_type);
+							return Err(());
+						},
+					};
+					debug!("Doing market order");
+
+					let new_market_order = NewMarketOrder {
+						request_id: intent_id,
+						chain_id: pumpx_config.to_chain_id,
+						token_ca: to_token_ca.clone(),
+						swap_type: match pumpx_config.swap_type {
+							1 => SwapType::Buy,
+							2 => SwapType::Sell,
+							_ => {
+								log::error!("Unsupported swap type: {}", pumpx_config.swap_type);
+								return Err(());
+							},
+						},
+						// amount received from binance should be used...
+						amount_in: to_amount.to_string(),
```

Comment:
Here we need to consider decimal, from previous example, backend expects `0.01` here, not `0.01 * (10^18)`

Reply:
Conversion to bnb unit (Jager) is done later. I expect current impl will result in `0.01` not `0.01 * 10^8`